<!DOCTYPE html>
<html lang="en" dir="ltr">

      <head>
        <meta charset="utf-8">
        <title>UniDL4BioPep</title>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family==Rancho&effect=shadow-multiple' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="{{ url_for('static', filename='css/style.css') }}">

		<style>
			.content {
			  max-width: 1000px;
			  margin: auto;
			}
            input[type=checkbox]{
                width: 20px;
            }
        </style>
    </head>

<body>
    <div class="content">
        <h1>UniDL4BioPep: A universal deep learning architecture for bioactive peptide prediction</h1>
        <p style="font-size:70%">The webserver is the implementation of the paper "Du, Z., Ding, X., Xu, Y., & Li, Y. (2023). UniDL4BioPep: a universal deep learning architecture for binary classification in peptide bioactivity. Briefings in Bioinformatics, bbad135."</p>
        <p style="font-size:70%">Notice: For very large dataset processing: please download our model locally or contact us at zhenjiao@ksu.edu or yonghui@ksu.edu for more assistant.  </p>
	<fieldset>
        <p style="font-size:80%">Quick output version: 1. Choose a model  → 2. Input a peptide sequence</p>
        <form action="{{url_for('predict')}}" method="post">
            <label for="Model selection">Choose a model:</label>
            <select name="Model_selection" id="model2" required="required">
                <option value="1">Antihypertensive</option>
                <option value="2">DPPIV</option>
                <option value="3">bitter</option>
                <option value="4">umami</option>
                <option value="5">AMP</option>
                <option value="6">AMAP_main</option>
                <option value="7">AMAP_alternative</option>
                <option value="8">QS</option>
                <option value="9">ACP_main</option>
                <option value="10">ACP_alternative</option>
                <option value="11">MRSA</option>
                <option value="12">TTCA</option>
                <option value="13">Antifungal</option>
                <option value="14">antioxidant</option>
                <option value="15">Antiviral</option>
                <option value="16">Antibacterial</option>
                <option value="17">BBP</option>
                <option value="18">Toxicity</option>
                <option value="19">Neuropeptide</option>
                <option value="20">APP</option>
                <option value="21">TTCA(FL)</option>
                <option value="22">umami(FL)</option>
                <option value="23">AMAP_alternative(FL)</option>
                <option value="24">AMAP_main(FL)</option>
                <option value="25">AMP(FL)</option>
                <option value="26">MRSA(FL)</option>

            </select>
{#            <input type="text" name="Model selection" placeholder="input model index, example:1" required="required">#}
            <input type="text" name="Peptide_sequences" placeholder="peptide sequence input, example: ASD,QWE,APE" required="required" />
            <button type="submit" class="btn btn-primary btn-block btn-large">Run</button>
        </form>
        <p style="color:cyan"> {{ prediction_text }} </p>
        </fieldset>
        <br/>
        <br/>
        <fieldset>
        <p style="font-size:80%">Large-scale output version: 1. Prepare your files (xls, xlsx, fasta, or txt) and click “Choose File” for uploading → 2. Choose one or multiple models  → 3. Download the results.</p>
        <form action="{{url_for('pred_with_file')}}" method="post" enctype = "multipart/form-data">
            <label for="Model_selection">Choose your favorite model(s):</label> <br/>
                <label><input type="checkbox" name="Model_selection" value="1" checked>Antihypertensive</label>
                <label><input type="checkbox" name="Model_selection" value="2">DPPIV</label>
                <label><input type="checkbox" name="Model_selection" value="3">bitter</label>
                <label><input type="checkbox" name="Model_selection" value="4">umami</label>
                <label><input type="checkbox" name="Model_selection" value="5">AMP</label>
                <label><input type="checkbox" name="Model_selection" value="6">AMAP_main</label>
                <label><input type="checkbox" name="Model_selection" value="7">AMAP_alternative</label>
                <label><input type="checkbox" name="Model_selection" value="8">QS</label>
                <label><input type="checkbox" name="Model_selection" value="9">ACP_main</label>
                <label><input type="checkbox" name="Model_selection" value="10">ACP_alternative</label>
                <label><input type="checkbox" name="Model_selection" value="11">MRSA</label>
                <label><input type="checkbox" name="Model_selection" value="12">TTCA</label>
                <label><input type="checkbox" name="Model_selection" value="13">Antifungal</label>
                <label><input type="checkbox" name="Model_selection" value="14">Antioxidant</label>
                <label><input type="checkbox" name="Model_selection" value="15">Antiviral</label>
                <label><input type="checkbox" name="Model_selection" value="16">Antibacterial</label>
                <label><input type="checkbox" name="Model_selection" value="17">BBP</label>
                <label><input type="checkbox" name="Model_selection" value="18">Toxicity</label>
                <label><input type="checkbox" name="Model_selection" value="19">neuropeptide</label>
                <label><input type="checkbox" name="Model_selection" value="20">APP</label>
                <label><input type="checkbox" name="Model_selection" value="21">TTCA(FL)</label>
                <label><input type="checkbox" name="Model_selection" value="22">umami(FL)</label>
                <label><input type="checkbox" name="Model_selection" value="23">AMAP_alternative(FL)</label>
                <label><input type="checkbox" name="Model_selection" value="24">AMAP_main(FL)</label>
                <label><input type="checkbox" name="Model_selection" value="25">AMP(FL)</label>
                <label><input type="checkbox" name="Model_selection" value="26">MRSA(FL)</label>


            <br/>
            <label for="Peptide_sequences">Choose a file to upload</label>
            <input type="file" name="Peptide_sequences" id="Peptide_sequences" accept=".xls,.xlsx,.txt,.fasta"/>
            <button type="submit" class="btn btn-primary btn-block btn-large">Run</button>
          </form>
          <h4>Usage of the webserver:</h4>
          <p style="font-size:80%"> Example for “Quick output version” : </p>
          <p style="font-size:70%">1. Select “Antihypertensive” model for antihypertensive activity prediction.   →   →  →  2. Insert a peptide or protein sequence, “VPP” →   →  →  3. Click “Run”→   →  → 4. The result will be returned in seconds below the “Run” button</p>
          <p style="font-size:70%">Notice: it also support multiple sequence at the same time. Just input as “VPP,IPP,CCL,AGR” (sequences are separated by comma, no space)</p>
          <p style="font-size:80%"> Example for “Large-scale output version:” : </p>
          <p style="font-size:70%">1. Prepare your xls, xlsx, txt or fasta files  →  →  → 2. Upload the file through “Choose File” botton  →   →  → 3. Select one or several models →   →  → 4. Click “Run” →   →  → 5. It will automatically download your results. </p>
          <p style="font-size:70%">Notice: File preparation should follow the examples under this repository https://github.com/dzjxzyd/UniDL4BioPep_webserver/tree/main/Example%20uploading%20files </p>
          <h4>Detailed explaination of the activity abbreviation</h4>
	  <p style="font-size:70%"> Antihypertensive： Angiotensin-converting enzyme inhibitory activity (main target in for hypertension); DPPIV: dipeptidyl peptidase IV (DPPIV) inhibitory activity (main target for diabetes); AMP: antimicrobial activity; AMAP: antimalarial activity (main and alternative is corresponding to two datasets; QS: quorum-sensing activity; ACP: anticancer activity: MRSA: anti-methicillin-resistant S. aureus strains activity); TTCA: tumor T cell antigens; BBP: blood-brain barrier peptide; APP: anti-parasitic activity; FL: is just an indicator of the Focal loss as loss function version, typically, we recommend the FL-version if available (for balanced datassets, we do not using FL for model generation, but you can try it based our our template tutorial in github) </p>	
	  <h4>The whole model architecture</h4>
          <img src="static/css/Whole architecture.png" alt="Whole architecture" width="950" height="650" style="float:left">
          <h4>Dataset for the those models</h4>
          <table>
          <thead>
            <tr>
              <th>Bioactivity</th>
              <th>Training dataset</th>
              <th>Test dataset</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td style="text-align: center; vertical-align: middle;">ACE inhibitory activity</td>
              <td style="text-align: center; vertical-align: middle;">913 positives and 913 negatives</td>
              <td style="text-align: center; vertical-align: middle;">386 positives and 386 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">DPP IV inhibitory activity</td>
              <td style="text-align: center; vertical-align: middle;">532 positives and 532 negatives</td>
              <td style="text-align: center; vertical-align: middle;">133 positives and 133 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Bitter</td>
              <td style="text-align: center; vertical-align: middle;">256 positives and 256 negatives</td>
              <td style="text-align: center; vertical-align: middle;">64 positives and 64 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Umami</td>
              <td style="text-align: center; vertical-align: middle;">112 positives and 241 negatives</td>
              <td style="text-align: center; vertical-align: middle;">28 positives and 61 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Antimicrobial activity</td>
              <td style="text-align: center; vertical-align: middle;">3876 positives and 9552 negatives</td>
              <td style="text-align: center; vertical-align: middle;">2584 positives and 6369 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Antimalarial activity</td>
              <td style="text-align: center; vertical-align: middle;">Main dataset (111 positives and 1708 negatives); alternative dataset (111 positives and 542 negatives)</td>
              <td style="text-align: center; vertical-align: middle;">Main dataset (28 positives and 427 negatives); alternative dataset (28 positives and 135 negatives)</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Quorum sensing activity</td>
              <td style="text-align: center; vertical-align: middle;">200 positives and 200 negatives</td>
              <td style="text-align: center; vertical-align: middle;">20 positives and 20 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Anticancer activity</td>
              <td style="text-align: center; vertical-align: middle;">Main dataset (689 positives and 689 negatives); alternative dataset (776 positives and 776 negatives)</td>
              <td style="text-align: center; vertical-align: middle;">Main dataset (172 positives and 172 negatives); alternative dataset (194 positives and 194 negatives)</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Anti-MRSA strains activity</td>
              <td style="text-align: center; vertical-align: middle;">118 positives and 678 negatives</td>
              <td style="text-align: center; vertical-align: middle;">30 positives and 169 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Tumor T cell antigens</td>
              <td style="text-align: center; vertical-align: middle;">470 positives and 318 negatives</td>
              <td style="text-align: center; vertical-align: middle;">122 positives and 75 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Blood-Brain Barrier</td>
              <td style="text-align: center; vertical-align: middle;">100 positives and 100 negatives</td>
              <td style="text-align: center; vertical-align: middle;">19 positives and 19 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Anti-parasitic activity</td>
              <td style="text-align: center; vertical-align: middle;">255 positives and 255 negatives</td>
              <td style="text-align: center; vertical-align: middle;">46 positives and 46 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Neuropeptide</td>
              <td style="text-align: center; vertical-align: middle;">1940 positives and 1940 negatives</td>
              <td style="text-align: center; vertical-align: middle;">485 positives and 485 negatives</td>
            </tr>
            <tr>
              <td style="text-align: center; vertical-align: middle;">Antibacterial activity</td>
              <td style="text-align: center; vertical-align: middle;">6583 positives and 6583 negatives</td>
              <td style="text-align: center; vertical-align: middle;">1695 positives and 1695 negatives</td>
            </tr>
              <td style="text-align: center; vertical-align: middle;">Antifungal activity</td>
              <td style="text-align: center; vertical-align: middle;">778 positives and 778 negatives</td>
              <td style="text-align: center; vertical-align: middle;">215 positives and 215 negatives</td>
            </tr>
            <tr>
            <td style="text-align: center; vertical-align: middle;">Antiviral activity</td>
              <td style="text-align: center; vertical-align: middle;">2321 positives and 2321 negatives</td>
              <td style="text-align: center; vertical-align: middle;">623 positives and 623 negatives</td>
            </tr>
            <tr>
            <td style="text-align: center; vertical-align: middle;">Toxicity</td>
              <td style="text-align: center; vertical-align: middle;">1642 positives and 1642 negatives</td>
              <td style="text-align: center; vertical-align: middle;">290 positives and 290 negatives</td>
            </tr>
            <tr>
            <td style="text-align: center; vertical-align: middle;">Antioxidant activity</td>
              <td style="text-align: center; vertical-align: middle;">582 positives and 541 negatives</td>
              <td style="text-align: center; vertical-align: middle;">146 positives and 135 negatives</td>
            </tr>
          </table>
      </div>
  </body>
  </html>
